packages/dds/tree/src/simple-tree/proxies.ts

```diff
@@ -742,6 +742,21 @@ export const mapStaticDispatchMap: PropertyDescriptorMap = {
 			}
 		},
 	},
+	forEach: {
+		value(
+			this: TreeMapNode,
+			callbackFn: (value: unknown, key: string, map: ReadonlyMap<string, unknown>) => void,
+			thisArg?: any,
+		): void {
+			if (thisArg === undefined) {
+				// We can't pass `callbackFn` to `FlexTreeMapNode` directly, or else the third argument ("map") will be a flex node instead of the proxy.
+				getFlexNode(this).forEach((v, k, _) => callbackFn(v, k, this), thisArg);
```

Comment:
Nit: To deduplicate a bit, and to potentially make future debugging a bit easier, it might be nice to extract the `getFlexNode(this)` call out to a variable assignment above the conditional.